apps/extension/src/ui/domains/Ramps/coinbase/useCoinbaseSellQuote.ts

```diff
@@ -0,0 +1,198 @@
+    queryFn: () => {
+      if (inputError) return inputError
+      if (!config || !token || !coinbaseToken) return null
+      return fetchCoinbaseSellQuote(
+        config.currencyCode,
+        config.amount,
+        coinbaseToken,
+        token.decimals,
+        minMaxAmount,
+      )
+    },
+    select: (res: FetchCoinbaseSellQuoteResult | null): RampsSellQuote | null => {
+      if (!res) return null
+      if (res.type === "error") return res
+      return res.data && token && config && coinbaseToken
+        ? {
+            type: "success",
+            fee: Number(res.data.coinbase_fee.value),
+            amountOut: Number(res.data.cashout_total.value),
+            tokenPrice:
+              Number(res.data.cashout_subtotal.value) / Number(res.data.sell_amount.value),
+            getRedirectUrl: (address: string) =>
+              getCoinbaseSellUrl(
+                res.data.cashout_total.currency,
+                res.data.sell_amount.value,
+                coinbaseToken.sellCurrency,
+                coinbaseToken.sellNetwork,
+                res.data.quote_id,
+                res.data.cashout_total.value,
+                address,
+              ),
+          }
+        : null
+    },
+    retry: false,
+  })
+}
+
+type CoinbaseTokenSpecs = { sellCurrency: string; sellNetwork: string }
+
+const useCoinbaseTokenSpecs = (tokenId: string | undefined) => {
+  const { data: coinbaseBuyOptions } = useCoinbaseSellOptions()
+  const token = useToken(tokenId)
+
+  return useMemo<CoinbaseTokenSpecs | null>(() => {
+    if (!token) return null
+
+    const item = coinbaseBuyOptions?.sell_currencies
+      .flatMap((c) => c.networks.map((n) => ({ id: c.id, symbol: c.symbol, ...n })))
+      .find((n) => {
+        if (isEvmToken(token) && n.chain_id === token.evmNetwork?.id) {
+          if (
+            token.type === "evm-erc20" &&
+            token.contractAddress.toLowerCase() === n.contract_address.toLowerCase()
+          )
+            return true
+          if (token.type === "evm-native" && !n.contract_address) return true
+        }
+
+        if (isSubToken(token) && n.name === token.chain?.id && n.symbol === token.symbol)
+          return true
+
+        return false
+      })
+
+    return item ? { sellCurrency: item.id, sellNetwork: item.name } : null
+  }, [coinbaseBuyOptions?.sell_currencies, token])
+}
+
+type FetchCoinbaseSellQuoteResult =
+  | { type: "success"; data: CoinbaseSellQuoteResponse }
+  | RampsQuoteError
+
+const fetchCoinbaseSellQuote = async (
+  currencyCode: string,
+  amountIn: number,
+  coinbaseToken: CoinbaseTokenSpecs,
+  decimals: number,
+  minMaxAmount: string | null,
+): Promise<FetchCoinbaseSellQuoteResult> => {
+  const body: CoinbaseSellQuoteRequest = {
+    cashoutCurrency: currencyCode,
+    paymentMethod: "FIAT_WALLET",
+    sellAmount: formatTokens(amountIn, decimals), // dont send "1e-9" ^^
```

Comment:
`BigNumber(value).toFixed` also solves this!

> Unlike Number.prototype.toFixed, which returns exponential notation if a number is greater or equal to 1021, this method will always return normal notation.

From https://mikemcl.github.io/bignumber.js/#toFix

But your `formatTokens` also seems to do the trick